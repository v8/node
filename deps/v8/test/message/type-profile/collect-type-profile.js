// Copyright 2016 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Flags: --type-profile --turbo

function test(param) {
  var my_var1 = param;
  var my_var2 = 17;
}

test({});
test(123);
test('hello');
test(123);
test(undefined);
test('hello');
test({x: 12});
test({x: 12});

class MyClass {
  constructor() {}
}


function testConstructorNames(param) {
  var my_var = param;
}

testConstructorNames(new MyClass());
testConstructorNames({});
testConstructorNames(2);

throw "throw otherwise test fails with --stress-opt";
