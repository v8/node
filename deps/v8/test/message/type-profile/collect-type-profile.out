my_var1
Object

my_var2
number

my_var1
Object
number

my_var2
number
number

my_var1
Object
number
string

my_var2
number
number
number

my_var1
Object
number
string
number

my_var2
number
number
number
number

my_var1
Object
number
string
number
undefined

my_var2
number
number
number
number
number

my_var1
Object
number
string
number
undefined
string

my_var2
number
number
number
number
number
number

my_var1
Object
number
string
number
undefined
string
Object

my_var2
number
number
number
number
number
number
number

my_var1
Object
number
string
number
undefined
string
Object
Object

my_var2
number
number
number
number
number
number
number
number


Function

MyClass
Function

my_var
MyClass

my_var
MyClass
Object

my_var
MyClass
Object
number

*%(basename)s:34: throw otherwise test fails with --stress-opt
throw "throw otherwise test fails with --stress-opt";
^
